torch/csrc/jit/import_source.h

```diff
@@ -10,28 +10,35 @@
 namespace torch {
 namespace jit {
 namespace script {
-// Helpers to define modules and classes from source strings. Used in model
-// saving/loading, so it expects the format produced by the model exporter.
-
-// Define the list of classes in `src`.
-TORCH_API void import_libs(
-    // The compilation unit that will own the imported libs
-    std::shared_ptr<CompilationUnit> cu,
-    // Qualifier for any classes that `src` defines. Looks like a module path,
-    // like "foo.bar.baz"
-    const std::string& class_qualifier,
-    const std::shared_ptr<Source>& src,
-    const std::vector<at::Tensor>& tensor_table,
-    // Callback to import any dependencies of this source before compiling
-    const std::function<void(const std::string&)>& import_callback);
-
-// Add the methods defined in `src` to the module `mod`.
-TORCH_API void LEGACY_import_methods(
-    const script::Module& mod,
-    const std::shared_ptr<Source>& src,
-    const std::vector<at::Tensor>& constant_table,
-    // Callback to import any dependencies of this source before compiling
-    const std::function<void(const std::string&)>& import_callback);
+
+struct SourceImporterImpl;
+
+// Given a directory of serialized TorchScript sources,
+// This class allows the loading of individual named types in source.
+// Resolves the dependencies between source files and parses
+// the source files as necessary.
+using SourceLoader = std::function<std::shared_ptr<Source>(const std::string&)>;
+
+struct TORCH_API SourceImporter {
```

Comment:
Can we just have `import.cpp` use a `Resolver` directly instead of having a SourceImporter? Could be nice as it would allow us to drop a concept.

Reply:
That is true, there is some weird API differences though: resolver takes a string and a source location. Providing the source location would be hard. I can yak shave further and change the resolver API though.